### DISCUSSION TITLE
Fix autoscaler issues

### DISCUSSION THREAD

**Discussion by xbeastx** [celery/celery#7034] [General]
We use autoscaler and it is greate featuer. Please return autoscaler code in 4.0

Why you has droped support this feature?

**Comment by timofurrer** [comment]
It seems like it's been removed because of lack of resources. 
See https://github.com/celery/celery/issues/3240 and https://github.com/celery/celery/commit/8e0ab0f09254039cdf05f18ff283bcbb0d5d99be

**Comment by ask** [comment]
We have a company that's interested in supporting this feature, but we have yet not come to a conclusion if autoscaling is the feature that they actually need.

**Comment by xbeastx** [comment]
@timofurrer wow, see this bug I leave comment there. Any other complaints about autoscaling?

**Comment by xbeastx** [comment]
@ask I think it must be not only about companies. It's about "big celery community" (as you wrote in some articles). And I have no any reasons to remove this feature. Because we use it in our project and it works (with some problems but works).

In our project we have long-running tasks which has different system load. So it was very useful to wrote our own autoscaler which increase count of workers when CPU/HDD/Memory usage is low and decrease count of workers otherwise (we can try share this autoscaler with community) And I have no idea how it's possible achieve this results with celery without this feature. And I think it's common problem (resources vs processing) in tasks processing systems that it must be useful for any kind of tasks in all projects.
Then probably we will trying write our own bicycle task processing framework with this feature but more simpler than celery.

You wrote somewhere that it will be undocumented but in current master branch it's code was deleted completely.

**Comment by gcarothers** [comment]
@ask you've got at least one other company that would be happy to sponsor the work as well! Use it for managing a somewhat spiky task load but that isn't worth (yet) dividing between more auto scaling instances (kubernetes style). Autoscale fits the "just right" balancing

**Comment by npilon** [comment]
> Use it for managing a somewhat spiky task load

Spiky _network I/O-heavy_ task load that sometimes yields big-ish resident memory sizes. We had been using `--autoscale 20,3` as a simple solution to processing lots of tasks quickly then not having a dozen idle workers with a significant amount of allocated memory between them.

**Comment by ask** [comment]
Autoscale code is returned...

We need to fix the bugs though,  @gcarothers I'll be in touch :)

**Comment by jcushman** [comment]
@gcarothers, @ask, is there any news on whether autoscaling is likely to get support to stick around as a feature?

**Comment by thedrow** [comment]
Not at the current moment

**Comment by auvipy** [comment]
can you please check 163d09f20a72550fb86e1787978291278e90b58a guys?

**Comment by auvipy** [comment]
> @gcarothers, @ask, is there any news on whether autoscaling is likely to get support to stick around as a feature?

please check 163d09f20a72550fb86e1787978291278e90b58a

**Comment by auvipy** [comment]
some improvements were done on this issue which are linked. but for further discussion moving this to forum.
